CMakeLists.txt

```diff
@@ -0,0 +1,923 @@
+include(${CMAKE_CURRENT_LIST_DIR}/predefine.cmake)
+
+################################# engine source code ##################################
+set(CC_COCOS_SOURCES)
+set(ENGINE_NAME cocos_engine)
+
+# Should be enable someday in the future
+# set(CMAKE_CXX_FLAGS "${WERROR_FLAGS}")
+
+################################# options ############################################
+option(CC_USE_GFX_RENDERER           "GFX Render"                       ON)
+option(CC_USE_VIDEO                "Enable VideoPlayer Component"       ON)
+option(CC_USE_WEBVIEW              "Enable WebView Component"           ON)
+option(CC_USE_AUDIO                "Enable Audio"                       ON)
+option(CC_USE_SOCKET               "Enable WebSocket & SocketIO"        ON)
+option(CC_USE_WEBSOCKET_SERVER     "Enable WebSocket Server"            OFF)
+option(CC_USE_MIDDLEWARE           "Enable Middleware"                  ON)
+option(CC_USE_SPINE                "Enable Spine"                       ON)
+option(CC_USE_DRAGONBONES          "Enable Dragonbones"                 ON)
+option(CC_USE_SE_NAPI          "Enable Dragonbones"                     ON)
+option(CC_USE_PARTICLE              "Enable Particle"                   ON)
+
+
+################################# external source code ################################
+set(EXTERNAL_ROOT ${CMAKE_CURRENT_LIST_DIR}/external)
+if(NOT EXISTS ${EXTERNAL_ROOT}/CMakeLists.txt)
+    message(FATAL_ERROR "Please download external libraries! File ${CMAKE_CURRENT_LIST_DIR}/external/CMakeLists.txt not exist!")
+endif()
+
+include_directories(${EXTERNAL_ROOT}/sources)
+include(${EXTERNAL_ROOT}/CMakeLists.txt)
+
+################################# cocos engine source code ################################
+
+##### cocos2d
+list(APPEND CC_COCOS_SOURCES
+    ${CMAKE_CURRENT_LIST_DIR}/cocos/cocos2d.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/cocos/cocos2d.h
+)
+
+##### platform
+list(APPEND CC_COCOS_SOURCES
```

Comment:
Why not use `cocos_source_files`?

Reply:
cocos_source_files is a func that is used in creator 3.x ,it is too complex for creator 2.x because cmake is only used for openharmony.